packages/wrangler/src/cfetch/internal.ts

```diff
@@ -74,7 +74,7 @@ function truncate(text: string, maxLength: number): string {
 function cloneHeaders(
   headers: HeadersInit | undefined
 ): Record<string, string> {
-  return { ...headers };
+  return Array.isArray(headers) ? Object.fromEntries(headers) : { ...headers };
```

Comment:
`headers: HeadersInit` could also be a `Headers` instance which won't properly be spread by `{ ...headers }`

Reply:
In our code, we never actually pass in a `Headers` object, so we kinda get away with this.
But I will make a better fix.